### ISSUE TITLE
Performance of pre-trained model

### ISSUE THREAD

**Issue by goyalsaransh97** [facebookresearch/pythia#54]
On using the pretrained pythia model downloaded, I'm just getting 66.7% overall accuracy on test-dev, which is much lower than reported single-model accuracy. Am I doing something wrong?
I tried downloading train+dev model from https://dl.fbaipublicfiles.com/pythia/pretrained_models/textvqa/pythia_train_val.pth
but the link seems broken.

Also, how much GPU memory is needed to train the model without any changes? I've a V100 but I'm getting not sufficient memory errors.
Thanks

**Comment by apsdehal** [comment]
The correct link is https://dl.fbaipublicfiles.com/pythia/pretrained_models/vqa2/pythia_train_val.pth. Apologies for the confusion. 

You wouldn't be able to fit whole 512 size batch on single v100. Decrease the batch size to 256. There would be minor performance loss, but not that much.

**Comment by goyalsaransh97** [comment]
Thanks for the updated link. How much accuracy should I expect for both the pretrained models?
I do have multiple V100s but Pythia doesn't seem to use them. Is there a simple solution to this? To train over multiple GPUs and retain original batch size?
For batch size of 256, how should the learning rate be modified?

**Comment by apsdehal** [comment]
See the configuration section of the documentation. Specifically, for running on multiple GPUS, add `training_parameters.data_parallel True` to the end of your command. Accuracies for both versions are mentioned in the README.

**Comment by goyalsaransh97** [comment]
Thanks for the quick reply

**Comment by goyalsaransh97** [comment]
In training of the model with 3 V100 GPUs, I'm getting CUDA memory error after 5000 iterations. Could you suggest something w.r.t. this?

**Comment by apsdehal** [comment]
This technically shouldn't be an issue as if it runs fine for 5000 iterations, it should run fine later on also. Can you post your log file here? Also, can try using even number of GPUs? (Use environment `CUDA_VISIBLE_DEVICES`)

Alternatively, you can try decreasing batch size to 256 and half the learning rate (but it can require more fine-tuning than that).

**Comment by apsdehal** [comment]
@goyalsaransh97 Should we close this?

**Comment by goyalsaransh97** [comment]
Does the Pythia pre-trained model contain grid-features and 100 b-boxes?
If yes, why is the test-dev score much lower than reported for single model? How can I push it to the reported ones?
If not, how can I train the model with and without the grid-features?

**Comment by apsdehal** [comment]
Yes, it contains both of them.
- What is the test-dev score you are getting? 
- Are you using and training the one that uses both `train+val`? 
- Which reported number are you referring to?

**Comment by goyalsaransh97** [comment]
- I'm getting 68.3, same as mentioned in the Readme.
- Yes
- Shouldn't the score be 70.0, as mentioned in the paper?

**Comment by apsdehal** [comment]
If you want to reproduce that number you will have to collect extra data from v0.1 tag. See https://github.com/facebookresearch/pythia/tree/5ad666f417684f8f1280029ac46e19dc8b5ddb0a. It contains extra data from visual genome as well as visual dialog and some augmentation data. We are not supporting that model with extra training data anymore in v0.3 and focusing on pure VQA training without additional data. 

In short, if you want to reproduce numbers present in the paper, you will have to use v0.1.
